### PR TITLE
SOLR support and search improvements

### PR DESCRIPTION
**Pull request by dynamicguy**
1. user search is now working in url /user-management
2. bulk operations like index sync, import, export and delete selected
3. added solr as an option beside elasticsearch as search engine
4. broken listing page header fix

here is a screenshot from my generated app: 
screenshot 2015-11-06 18 09 42

and here is that projects source code: https://github.com/dynamicguy/jhipster-solr-example

### CODE REVIEW

File: app/templates/_bower.json [hunk 1]
```diff
@@ -4,19 +4,19 @@
     "appPath": "src/main/webapp",
     "testPath": "src/test/javascript/spec",
     "dependencies": {
-        "angular": "1.4.5",
-        "angular-aria": "1.4.5",
+        "angular": "1.4.x",
```
Comment (deepu105) [inline app/templates/_bower.json]: change to proper versions as per our policy

File: app/templates/_package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "<%= slugifiedBaseName %>",
-  "version": "0.0.0",
-  "description": "Description for <%= baseName %>",
+  "version": "1.0.0",
+  "description": "<%= baseName %>",
```
Comment (deepu105) [inline app/templates/_package.json]: these changes should not be here

File: app/templates/src/main/scss/main.scss [hunk 3]
```diff
@@ -12,6 +12,27 @@ body {
...
+    background-color: #fcf8e3;
+    padding: 10px 15px;
+    border: 1px solid #faebcc;
+}
+
+.caption {
+    margin: 10px 0;
+}
+
+.table > thead > tr > th, .table > thead > tr > td, .table > tbody > tr > th, .table > tbody > tr > td, .table > tfoot > tr > th, .table > tfoot > tr > td {
```
Comment (deepu105) [inline app/templates/src/main/scss/main.scss]: we use bootstrap and hence should stick to bootstrap css

File: app/templates/src/main/webapp/_index.html [hunk 4]
```diff
@@ -1,6 +1,32 @@
...
     <head>
+	    <!--
+	        IE8 support, see AngularJS Internet Explorer Compatibility http://docs.angularjs.org/guide/ie
+	        For Firefox 3.6, you will also need to include jQuery and ECMAScript 5 shim
+	    -->
+	    <!--[if lt IE 9]>
+	    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.js"></script>
+	    <script src="http://cdnjs.cloudflare.com/ajax/libs/es5-shim/2.2.0/es5-shim.js"></script>
+	    <script>
+	        document.createElement('ui-select');
```
Comment (deepu105) [inline app/templates/src/main/webapp/_index.html]: we do not use ui-select

File: app/templates/src/main/webapp/_index.html [hunk 5]
```diff
@@ -1,6 +1,32 @@
 <!doctype html>
-<html class="no-js">
+<html class="no-js" ng-app="<%= angularAppName %>" xmlns:ng="http://angularjs.org">
```
Comment (deepu105) [inline app/templates/src/main/webapp/_index.html]: ng-app declared twice

File: app/templates/src/main/webapp/i18n/en/_global.json [hunk 6]
```diff
@@ -82,7 +82,12 @@
             "delete": "Delete",
             "edit": "Edit",
             "save": "Save",
-            "view": "View"
+            "view": "View",
+			"found": "[<b>{{found}}</b>] items retrieved from total of [<b>{{total}}</b>] items.",
```
Comment (deepu105) [inline app/templates/src/main/webapp/i18n/en/_global.json]: format

File: app/templates/src/main/webapp/scripts/app/admin/audits/audits.html [hunk 7]
```diff
@@ -1,8 +1,7 @@
 <div>
-    <h2 translate="audits.title">Audits</h2>
-
     <div class="row">
-        <div class="col-md-5">
+		<div class="col-md-5"><h2 translate="audits.title">Audits</h2></div>
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/audits/audits.html]: format

File: app/templates/src/main/webapp/scripts/app/admin/metrics/_metrics.html [hunk 8]
```diff
@@ -41,25 +47,27 @@ <h3 translate="metrics.jvm.title">JVM Metrics</h3>
...
-        <div class="row">
-            <div class="col-md-9" translate="metrics.jvm.gc.scavengecount">Scavenge count</div>
-            <div class="col-md-3 text-right">{{metrics.gauges['jvm.garbage.PS-Scavenge.count'].value}}</div>
-        </div>
-        <div class="row">
-            <div class="col-md-9" translate="metrics.jvm.gc.scavengetime">Scavenge time</div>
-            <div class="col-md-3 text-right">{{metrics.gauges['jvm.garbage.PS-Scavenge.time'].value}}ms</div>
-        </div>
-    </div>
+
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/metrics/_metrics.html]: format???

File: app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.controller.js [hunk 9]
```diff
@@ -38,7 +39,76 @@ angular.module('<%=angularAppName%>')
...
+            }), function (response) {
+                if (response.status === 404) {
+                    $scope.loadAll();
+                }
+            });
+        };
+
+		$scope.areAllUsersSelected = false;
+
+        $scope.updateUsersSelection = function (userArray, selectionValue) {
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.controller.js]: format

File: app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.controller.js [hunk 10]
```diff
@@ -38,7 +39,76 @@ angular.module('<%=angularAppName%>')
...
+                    //TODO: handle bulk import
+                }
+            }
+        };
+
+        $scope.export = function (){
+            for (var i = 0; i < $scope.users.length; i++){
+                var user = $scope.users[i];
+                if(user.isSelected){
+                    //User.update(user);
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.controller.js]: incomplete??

File: app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.controller.js [hunk 11]
```diff
@@ -38,7 +39,76 @@ angular.module('<%=angularAppName%>')
...
+                    //TODO: handle bulk export
+                }
+            }
+        };
+
+        $scope.deleteSelected = function (){
+            for (var i = 0; i < $scope.users.length; i++){
+                var user = $scope.users[i];
+                if(user.isSelected){
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/_user-management.controller.js]: incomplete??

File: app/templates/src/main/webapp/scripts/app/admin/user-management/user-management.html [hunk 12]
```diff
@@ -30,15 +76,15 @@ <h2 class="col-sm-6" translate="user-management.home.title">Users</h2>
                     <span class="label label-success" ng-click="setActive(user, false)" ng-show="user.activated"
                           translate="user-management.activated" style="cursor: pointer;">Activated</span>
                 </td>
-                <% if (enableTranslation) { %><td>{{user.langKey}}</td><% } %>
+                <td>{{user.langKey}}</td>
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/user-management.html]: this should not be removed

File: app/templates/src/main/webapp/scripts/app/admin/user-management/user-management.html [hunk 13]
```diff
@@ -30,15 +76,15 @@ <h2 class="col-sm-6" translate="user-management.home.title">Users</h2>
...
                 <td>
                     <div ng-repeat="authority in user.authorities">
                         <span class="label label-info">{{ authority }}</span>
                     </div>
-                </td><% if (databaseType == 'sql' || databaseType == 'mongodb') { %>
```
Comment (deepu105) [inline app/templates/src/main/webapp/scripts/app/admin/user-management/user-management.html]: why is this removed

File: entity/templates/src/main/webapp/app/_entity-controller.js [hunk 14]
```diff
@@ -87,4 +88,65 @@ angular.module('<%=angularAppName%>')
...
+                    //TODO: handle bulk export
+                }
+            }
+        };
+
+        $scope.export = function (){
+            for (var i = 0; i < $scope.<%= entityInstance %>s.length; i++){
+                var <%= entityInstance %> = $scope.<%= entityInstance %>s[i];
+                if(<%= entityInstance %>.isSelected){
+                    //<%= entityClass %>.update(<%= entityInstance %>);
```
Comment (deepu105) [inline entity/templates/src/main/webapp/app/_entity-controller.js]: incomplete??

File: test/test-creation.js [hunk 15]
```diff
@@ -200,6 +200,7 @@ describe('JHipster generator ', function () {
             'src/main/webapp/scripts/components/navbar/navbar.html',
             'src/main/webapp/scripts/components/navbar/navbar.controller.js',
             'src/main/webapp/scripts/components/user/user.service.js',
+			'src/main/webapp/scripts/components/user/user.search.service.js',
```
Comment (deepu105) [inline test/test-creation.js]: format

**Comment by deepu105** [comment]
Not sure if we can accept this PR, as there are lot of issues in this, look at some of my comments, formatting is off in most places, some of the functions are incomplete, we cant have it
though i would like to have these features but it should be working if we are generating it and not just method skeletons
@dynamicguy please also change the PR title properly

**Comment by deepu105** [comment]
the Travis build is broken as well

**Comment by jdubois** [comment]
This are all really cool-looking features, but we just can't merge something that big.
Could you have a look at the guidelines ? We need to discuss stuff before, and then have one PR per feature.
We all do this on our free time, so in the end this kind of big PR is not manageable for us.

**Comment by atomfrede** [comment]
Maybe we should split it into the parts already mentioned by @ferdous 

* PR 1: user search is now working in url /user-management
* PR 2: bulk operations like index sync, import, export and delete selected
* PR 3: broken listing page header fix
 and finally 
* PR 4: Add Solr as an option for the search backend (Or we use it as a proof of concept for the plugin system @jdubois )

**Comment by ferdous** [comment]
I would love to do further modification if required. This is very elegant generator. But without modification it looks so bad. beside the the aforementioned features I also pushed "sorting" in listing fields. 

If you guys agree I can fix those issues you raised in that PR.

Thanks
-- 
Nurul Ferdous
Sent with Airmail

**Comment by deepu105** [comment]
@ferdous you are welcome if you can split into seperate PRs as mentioned by @atomfrede ? Also plz take note of the line comments that we have left in this PR

**Comment by deepu105** [comment]
Closing this since we all agree on smaller PRs

**Pull request closed without merging**